util/src/test/java/tech/pegasys/teku/util/async/SafeFutureTest.java

```diff
@@ -502,6 +503,201 @@ public void allOfFailFast_completesWhenAllFuturesComplete() {
     assertThat(result).isCompleted();
   }
 
+  private static boolean hasDependents(CompletableFuture<?> fut) {
+    return fut.toString().toLowerCase().contains("dependents");
+  }
+
+  @Test
+  public void orTest1() throws Exception {
+    SafeFuture<Integer> fut0 = new SafeFuture<>();
+    SafeFuture<Integer> fut1 = new SafeFuture<>();
+
+    assertThat(hasDependents(fut0)).isFalse();
+
+    SafeFuture<Integer> orFut = fut0.or(fut1);
+
+    assertThat(hasDependents(fut0)).isTrue();
+    assertThat(hasDependents(fut1)).isTrue();
+
+    fut1.complete(12);
+
+    assertThat(hasDependents(fut0)).isFalse();
+    assertThat(hasDependents(fut1)).isFalse();
+    assertThat(orFut.get()).isEqualTo(12);
+  }
+
+  @Test
+  public void orTest2() throws Exception {
+    SafeFuture<Integer> fut0 = new SafeFuture<>();
+    SafeFuture<Integer> fut1 = new SafeFuture<>();
+
+    assertThat(hasDependents(fut0)).isFalse();
+
+    SafeFuture<Integer> orFut = fut0.or(fut1);
+
+    assertThat(hasDependents(fut0)).isTrue();
+    assertThat(hasDependents(fut1)).isTrue();
+
+    fut0.complete(12);
+
+    assertThat(hasDependents(fut0)).isFalse();
+    assertThat(hasDependents(fut1)).isFalse();
+    assertThat(orFut.get()).isEqualTo(12);
+  }
+
+  @Test
+  public void orTest3() {
+    SafeFuture<Integer> fut0 = new SafeFuture<>();
+    SafeFuture<Integer> fut1 = new SafeFuture<>();
+
+    assertThat(hasDependents(fut0)).isFalse();
+
+    SafeFuture<Integer> orFut = fut0.or(fut1);
+
+    assertThat(hasDependents(fut0)).isTrue();
+    assertThat(hasDependents(fut1)).isTrue();
+
+    fut1.completeExceptionally(new IllegalStateException());
+
+    assertThat(hasDependents(fut0)).isFalse();
+    assertThat(hasDependents(fut1)).isFalse();
+    assertThat(orFut).isCompletedExceptionally();
+  }
+
+  @Test
+  public void interruptTest1() throws Exception {
```

Comment:
(nit) The generally format we tend to use for unit tests is <methodBeingTested_testCase>, would be nice to update these test names to be more descriptive.  For example: 
```suggestion
  public void orInterrupt_completeWithoutFiringInterrupt() throws Exception {
```